src/components/MultiHopTrade/hooks/useIsApprovalNeeded.tsx

```diff
@@ -0,0 +1,66 @@
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { useCallback, useEffect, useState } from 'react'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { usePoll } from 'hooks/usePoll/usePoll'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bn } from 'lib/bignumber/bignumber'
+import type { TradeQuote } from 'lib/swapper/api'
+import { erc20AllowanceAbi } from 'lib/swapper/swappers/utils/abi/erc20Allowance-abi'
+import { getERC20Allowance } from 'lib/swapper/swappers/utils/helpers/helpers'
+import { getWeb3InstanceByChainId } from 'lib/web3-instance'
+
+const APPROVAL_CHECK_INTERVAL_MILLISECONDS = 10_000
```

Comment:
Something to potentially revisit in the future, for EVM chains like Gnosis, we could get away with smaller interval checks